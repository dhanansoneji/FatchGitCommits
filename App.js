/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, {useEffect, useState} from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
} from 'react-native';

import {create} from 'apisauce';

import {Colors} from 'react-native/Libraries/NewAppScreen';

const api = create({
  baseURL: 'https://api.github.com',
  headers: {Accept: 'application/vnd.github.v3+json'},
});

const App = () => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  const [commits, setCommits] = useState([]);

  const getCommits = () => {
    api
      .get('/repos/dhanansoneji/Cats-Care/commits')
      .then(response =>
        response.data.map(e => {
          return {
            commitSHA: e.sha,
            author: e.commit.author.name,
            message: e.commit.message,
          };
        }),
      )
      .then(response => {
        setCommits(response);
        console.log(response);
      });
  };

  useEffect(() => {
    getCommits();
  }, []);

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <ScrollView contentInsetAdjustmentBehavior="automatic">
        <View
          style={{
            backgroundColor: isDarkMode ? Colors.black : Colors.white,
          }}
        />
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({});

export default App;
