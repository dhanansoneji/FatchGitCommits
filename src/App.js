/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React, {useEffect, useState} from 'react';
import {
  FlatList,
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
  Pressable,
  Linking,
} from 'react-native';

import Header from './components/MyHeader';

import {create} from 'apisauce';

import {Colors} from 'react-native/Libraries/NewAppScreen';

const api = create({
  baseURL: 'https://api.github.com',
  headers: {Accept: 'application/vnd.github.v3+json'},
});

const App = () => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
    flex: 1,
  };

  const [commits, setCommits] = useState([]);

  const getCommits = () => {
    api
      .get('/repos/dhanansoneji/FetchGitCommits/commits')
      .then(response =>
        response.data.map(e => {
          return {
            commitSHA: e.sha,
            author: e.commit.author.name,
            message: e.commit.message,
            url: e.html_url,
          };
        }),
      )
      .then(response => {
        setCommits(response);
        console.log(response);
      });
  };

  useEffect(() => {
    getCommits();
  }, []);

  const keyExtractor = (item, index) => index.toString();

  const loadInBrowser = url => {
    Linking.openURL(url).catch(err => console.error("Couldn't load page", err));
  };

  const renderItem = ({item}) => (
    <Pressable
      style={[
        {backgroundColor: isDarkMode ? Colors.black : Colors.white},
        styles.listItem,
      ]}
      onPress={() => loadInBrowser(item.url)}>
      <Text style={styles.textAuthor}>@{item.author}</Text>
      <Text style={[styles.textSHA, styles.marginTop]}>{item.commitSHA}</Text>
      <Text
        style={[
          {
            color: isDarkMode ? Colors.white : Colors.black,
          },
          styles.marginTop,
        ]}>
        {item.message}
      </Text>
    </Pressable>
  );

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <Header title="Fetch Git Commits" isDarkMode={isDarkMode} />
      <View style={styles.marginTop} />
      <FlatList
        contentInsetAdjustmentBehavior="automatic"
        keyExtractor={keyExtractor}
        data={commits}
        renderItem={renderItem}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  marginTop: {
    marginTop: 10,
  },
  listItem: {
    paddingHorizontal: 20,
    paddingVertical: 10,
    marginBottom: 10,
    marginHorizontal: 10,
    borderRadius: 10,
  },
  textAuthor: {
    color: '#00FF00',
    fontWeight: 'bold',
  },
  textSHA: {
    color: '#0096FF',
  },
});

export default App;
