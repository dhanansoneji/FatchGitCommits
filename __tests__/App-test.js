/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../src/App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {render, FindAllReturn} from '@testing-library/react-native';

jest.useFakeTimers();
it('renders correctly', () => {
  expect(renderer.create(<App />)).toMatchSnapshot();
});

it('children length', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree.children.length).toBe(3);
});

it('renders default elements', () => {
  const {getAllByText} = render(<App />);

  expect(getAllByText('Fetch Git Commits').length).toBe(1);
});
