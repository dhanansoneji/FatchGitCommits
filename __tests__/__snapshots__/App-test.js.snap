// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "elevation": 5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 30,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                ÔåÅ
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontSize": 22,
              "fontWeight": "bold",
            }
          }
          text="Fetch Git Commits"
        >
          Fetch Git Commits
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  />
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</RCTSafeAreaView>
`;
